lib/tasks/storage_provider.rake

```diff
@@ -1,155 +1,57 @@
     end
-    Rails.logger.level = 3
-    Container.auditing_enabled = false
-    storage_provider = StorageProvider.default
-    Project.all.each do |project|
-      project.data_files.each do |data_file|
-        upload = data_file.upload
-        #versions
-        if upload
-          upload.chunks.all.each do |chunk|
-            begin
-              storage_provider.delete_object(project.id, chunk.object_path)
-              chunk.destroy
-              print "c."
-            rescue StorageProviderException => e
-              puts "#{chunk.sub_path} object could not be deleted #{e.message}"
-            end
-          end
-          begin
-            storage_provider.delete_object_manifest(project.id, upload.object_path)
-            upload.destroy if upload
-            print "u."
-          rescue StorageProviderException => e
-            puts "#{upload.sub_path} manifest could not be deleted #{e.message}"
-          end
-        end
-        data_file.destroy
-        print "d."
-      end
-      begin
-        storage_provider.delete_container(project.id)
-        print "p."
-      rescue StorageProviderException => e
-        puts "#{project.id} container could not be deleted #{e.message}"
-      end
-    end
-    puts "\n"
-    puts storage_provider.get_account_info.to_json
   end
+end
 
-  desc 'print storage_provider usage information'
-  task usage: :environment do
-    Rails.logger.level = 3
-    sp = StorageProvider.default
-    puts sp.get_account_info.to_json
-  end
-
-  desc 'destroys containers, manifests, and objects orphaned from destroyed or logically_deleted objects'
-  task prune: :environment do
-    Rails.logger.level = 3
-    to_prune = {
-      projects: [],
-      uploads: [],
-      chunks: []
-    }
-
-    storage_provider = StorageProvider.default
-    storage_provider.get_containers.each do |container|
-      unless Project.where(id: container).exists?
-        to_prune[:projects] << container
-      end
-
-      storage_provider.get_container_objects(container).each do |object|
-        begin
-          if storage_provider.get_object_metadata(container, object)["x-static-large-object"]
-            unless Upload.where(id: object).exists?
-              to_prune[:uploads] << [container, object]
-            end
-          else
-            upload_id, chunk_number = object.split('/')
-            unless Chunk.where(upload_id: upload_id, number: chunk_number).exists?
-              to_prune[:chunks] << [container, object]
-            end
-          end
-        rescue StorageProviderException => e
-          puts "container #{container} object #{object} returned by get_container_objects, but metadata not accessible? #{e.message}"
-        end
-      end
+namespace :storage_provider do
+  desc "creates a storage_provider using ENV"
+  task create: :environment do
+    unless ENV['STORAGE_PROVIDER_TYPE']
+      $stderr.puts "YOU MUST SET ENV[STORAGE_PROVIDER_TYPE]"
```

Comment:
I think it would be useful to return the list of supported storage provider types in this error message.